### PR TITLE
explain typescript and simper javascript code

### PR DESCRIPTION
**Pull request by ssured**
when people are not comfortable with es6, reading the model-definition code first time will (probably) raise extra questions for them. Later the readme can note there is more concise syntax possible, but that syntax can lead to typing problems, which I added as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,8 +277,44 @@ It is perfectly fine to chain multiple `views`, `props` calls etc in arbitrary o
 
 It is also possible to define lifecycle hooks in the _actions_ object, these are actions with a predefined name that are run at a specific moment. See [Lifecycle hooks](#lifecycle-hooks-for-typesmodel).
 
-_Tip: `(self) => ({ action1() { }, action2() { }})` is ES6 syntax for `function (self) { return { action1: function() { }, action2: function() { } }}`, in other words; it's short way of directly returning an object literal.
-For that reason a comma between each member of a model is mandatory, unlike classes which are syntactically a totally different concept._
+TypeScript:
```
Comment (mweststrate) [inline README.md]: Could you move the typescript not bit down, to the typescript section below in the README (or institute a separate page for TS in the docs/ folder and link to it). I fear having the TS explanation on the type is quite overwhelming and not relevant for everybody

File: README.md [hunk 2]
```diff
@@ -270,8 +277,44 @@ It is perfectly fine to chain multiple `views`, `props` calls etc in arbitrary o
...
+
+TypeScript is supported, but you sometimes need to take into account where your typings are available.
+
+The code below will not compile: TypeScript will complain that `self.upperProp` is not a property. Computed properties are only available after `.views` is evaluated.
+
+```typescript
+const Example = types.model('Example', { prop: types.string }).views(self => ({
```
Comment (mweststrate) [inline README.md]: Note that another alternative worth mentioning is, as long as there are no circular deps:

```javascript
const Example = types
  .model('Example', { prop: types.string })
  .views(self => ({
    get upperProp(): string {
      return self.prop.toUpperCase();
    }
  }))
  .views(self => ({
    get twiceUpperProp(): string {
      return self.upperProp + self.upperProp;
    }
  }));
```

**Comment by ssured** [comment]
@mweststrate The changes you requested are now updated in the readme. Hope this code can be merged; converage decrease seems to be a regression in the CI system.

**Comment by mweststrate** [comment]
Thanks! Merged

**Pull request merged**